core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -21,35 +21,55 @@
 import java.nio.ByteBuffer;
 import java.security.SecureRandom;
 import java.util.Base64;
+import java.util.Collection;
+import java.util.List;
 import java.util.Map;
 import org.apache.iceberg.CatalogProperties;
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
 import org.apache.iceberg.io.SeekableInputStream;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.util.ByteBuffers;
 
 public class StandardEncryptionManager implements EncryptionManager {
-  public static final int KEK_ID_LENGTH = 16;
-
-  private final transient KeyManagementClient kmsClient;
   private final String tableKeyId;
   private final int dataKeyLength;
   private final long writerKekTimeout;
-  private Map<String, KeyEncryptionKey> kekCache;
+
+  // a holder class of metadata that is not available after serialization
+  private static class KeyManagementMetadata {
```

Comment:
I added this so that all of the transient state is in a single reference. I think it could have a better name if you can think of one!